### PR TITLE
Use "strict" boundaries by default for adaptive resampling

### PR DESCRIPTION
**Pull request by svank**
Split from #279

After the discussion around defaults changes, I'm separating out this commit as well. After adding a menu of boundary-handling options in #279, the old behavior can be had with the "ignore" option, and a new "strict" option gives `NaN`s for any output pixels that aren't fully determined by the input data (which generally has the effect of trimming the edges of the output image if the input image doesn't fully span the output plane). The "strict" option seems like the safest default, but I don't feel too strongly if changing this default is more work than it's worth.

Until things are merged and rebased, "Set default boundary mode to 'strict' for adaptive algo" is the only commit belonging to this PR.

### CODE REVIEW

**Comment by astrofrog** [comment]
@svank - sorry for the delay in getting back to these PRs - I've now merged #276 so could you rebase this? (and definitely add an entry in ``CHANGES.rst`` for this PR). We can then wait until after the next release to actually merge this change.

Is there a warning yet in ``main`` about this upcoming change?

**Review by astrofrog — CHANGES_REQUESTED** [review]
@svank - sorry for the delay in getting back to these PRs - I've now merged #276 so could you rebase this? (and definitely add an entry in ``CHANGES.rst`` for this PR). We can then wait until after the next release to actually merge this change.

Is there a warning yet in ``main`` about this upcoming change?

**Comment by svank** [comment]
To avoid a bunch of merge conflicts, I'll close this PR and move this commit to the same PR that changes the default kernel.

**Pull request closed without merging**
